core/transaction/payload/RegisterName.go

```diff
@@ -0,0 +1,78 @@
+package payload
+
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/nknorg/nkn/common/serialization"
+	"github.com/nknorg/nkn/crypto"
+	. "github.com/nknorg/nkn/errors"
+)
+
+type RegisterName struct {
+	Registrant *crypto.PubKey
```

Comment:
Looks use programHash here is more reasonable, since raw public key is heavy for database.

Reply:
@doraemon0 We must use public key, because name is resolved to public key and relay is done with using public key (see `server/relay.go`). We can't get public key from program hash.

Reply:
How about encoded public key?

Reply:
@doraemon0 I think you should always use full public key because it's used as key in Chord always in full form. And Chord doesn't know anything about it being public key, so for Chord it's just a bunch of bytes. @yilunzhang please correct me if I'm wrong

Reply:
Chord is using encoded public key. When need to verify something, such as signature chain, decoded it to raw public key first.

Reply:
Done. Can resolve?